main.py

```diff
@@ -0,0 +1,546 @@
+        tar.add(source_dir, arcname=Path(source_dir).name)
+
+
+def process_jobs(
+    job_docs: list[JobDoc],
+    startime: datetime,
+    args,
+    log_configurer: Callable,
+    log_queue: Queue,
+    ql,
+) -> None:
+    """
+    Parses the given job docs with the appropriate method
+
+    Test Job IDs:
+    [
+        {"id": "JOB-TEST:V01:METAR:CTC:CEILING:MODEL:OPS"},
+        {"id": "JOB-TEST:V01:METAR:CTC:VISIBILITY:MODEL:OPS"},
+        {"id": "JOB-TEST:V01:METAR:GRIB2:MODEL:HRRR"},
+        {"id": "JOB-TEST:V01:METAR:GRIB2:MODEL:RAP_OPS_130"},
+        {"id": "JOB-TEST:V01:METAR:NETCDF:OBS"}
+    ]
+
+    Example job doc:
+    {
+        'id': 'JOB:V01:METAR:CTC:CEILING:MODEL:HRRR_RAP_130',
+        'name': 'job:v01:metar:ctc:ceiling:model:hrrr_rap_130',
+        'offset_minutes': 0,
+        'run_priority': 6,
+        'sub_type': 'ctc'
+    }
+    """
+    logger.info("Processing the job docs")
+
+    success_count = 0
+    fail_count = 0
+    for job in job_docs:
+        logger.info(f"Processing job: {job}")
+        # translate _ to __ and : to _ for the name field
+        # TODO - why is this needed?
+        name = job["name"].replace("_", "__").replace(":", "_")
+
+        # Add a logging file handler with a unique name for just this job
+        logpath = (
+            args.log_dir / f"{name}-{startime.strftime('%Y-%m-%dT%H:%M:%S%z')}.log"
+        )
+        f_handler = log_config.add_logfile(ql, logpath)
+
+        hostname = os.uname().nodename.split(".")[0]
+        metric_name = f"{name}_{hostname}"
+        logger.info(f"metric_name {metric_name}")
+
+        # create an output directory with the time this job was started.
+        output_dir = (
+            Path(args.output_dir)
+            / f"{job['sub_type']}_to_cb"
+            / "output"
+            / f"{startime.strftime('%Y%m%d%H%M%S')}"
+        )
+        create_dirs([output_dir])
+
+        num_processes = determine_num_processes()
+
+        config = {
+            "job_id": job["id"],
+            "credentials_file": str(args.credentials_file),
+            "output_dir": str(output_dir),
+            "threads": num_processes,
+        }
+        job_succeeded = False
+        match job["sub_type"]:
+            case "grib2":
+                # FIXME: Update calling code to raise instead of calling sys.exit
+                try:
+                    grib_ingest = grib2_to_cb.run_ingest_threads.VXIngest()
+                    grib_ingest.runit(
+                        config,
+                        log_queue,
+                        log_configurer,
+                    )
+                except SystemExit as e:
+                    if e.code == 0:
+                        # Job succeeded
+                        job_succeeded = True
+                else:
+                    job_succeeded = True
+            case "netcdf":
+                # FIXME: Update calling code to raise instead of calling sys.exit
```

Comment:
good idea. It should just raise.

Reply:
Thanks! It makes a lot of sense when each `run_ingest_threads.py` is its own entrypoint being called by Bash. However, now that we have the Python entrypoint, I think we could switch it to just raising some semantic error. I'd want to address it later in another PR.